src/library_glfw.js

```diff
@@ -996,6 +996,11 @@ var LibraryGLFW = {
           stencil: (GLFW.hints[0x00021006] > 0),   // GLFW_STENCIL_BITS
           alpha: (GLFW.hints[0x00021004] > 0)      // GLFW_ALPHA_BITS 
         }
+#if OFFSCREEN_FRAMEBUFFER
+        // TODO: Make GLFW explicitly aware of whether it is being proxied or not, and set these to true only when proxying is being performed.
+        contextAttributes.renderViaOffscreenBackBuffer = true;
```

Comment:
this combination of 2 lines appears 3 times. perhaps we should have a shared function for it?

or can those lines go in createContext which is called by them?

Reply:
Hmm, I think that would make the field initialization more cumbersome to read, since I'd still want to #if calls to that function out, and having an extra function, say,

```javascript
#if OFFSCREEN_FRAMEBUFFER
GL.enableOffscreenBackBuffer = function(attributes) {
  attributes.renderViaOffscreenBackBuffer = true;
  attributes.preserveDrawingBuffer = true;
}
#endif
...

glfwCreateWindow: function() {
  ...
#if OFFSCREEN_FRAMEBUFFER
  GL.enableOffscreenBackBuffer(contextAttributes);
#endif
...
}
```

builds an indirection that probably does not save much, and adds a mental detour when reading the code.

The semantical idea of `-s OFFSCREEN_FRAMEBUFFER=1` follows the same idea that `-s USE_WEBGL2=1` also does:
 - At compile-time, enable generating in the code to add the feature,
 - For APIs that are not able to express initialization parameters between Offscreen Framebuffer or not, always enable the feature when creating the context (GLUT, GLFW, SDL),
 - For HTML5 API, in which we are able to express initialization parameters for enabling Offscreen Framebuffer or not, don't always-enable the feature, but let user's submit the desired initialization mode in the context init attribute parameters.

This means that we would not want to make `createContext` unconditionally enable these bits for all created contexts if `-s OFFSCREEN_FRAMEBUFFER=1`, but let the caller decide.

(Also duplicating the code between SDL, GLUT and GLFW will not be a duplication in generated code, since those APIs are generally never used at the same time)

Reply:
Agreed on createContext, thanks.

I see what you're saying about the code duplication issue, but on the other hand, it would make the code more maintainable to use a little helper. It's likely we'll have more work done on these APIs, and having to remember to update all the various locations is risky.

Reply:
Gotcha, updated the code, how does it look now?